### PR TITLE
Added missing tests + Fix Coaching/Crafty Shield interaction

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
Also fixed KNOWN_FAILING_PASS not showing the correct test's line

## Images
Before and after
image
image

## **Discord contact info**
AsparagusEduardo

### CODE REVIEW

File: test/battle/move_effect/coaching.c [hunk 1]
```diff
@@ -1,8 +1,112 @@
 #include "global.h"
 #include "test/battle.h"
```
Comment (hedara90) [inline test/battle/move_effect/coaching.c:3]:
```suggestion

ASSUMPTIONS
{
    ASSUME(gMovesInfo[MOVE_COACHING].effect == EFFECT_COACHING);
}

```
Should have the assumption for Coaching effect.

File: test/battle/move_effect/coaching.c [hunk 2]
```diff
@@ -1,8 +1,112 @@
...
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_COACHING, playerLeft);
+        MESSAGE("Wynaut's Attack rose!");
+        MESSAGE("Wynaut's Defense rose!");
+    }
+}
+
+DOUBLE_BATTLE_TEST("Coaching bypasses Protect")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
```
Comment (hedara90) [inline test/battle/move_effect/coaching.c:23]:
```suggestion
    GIVEN {
        ASSUME(gMovesInfo[MOVE_PROTECT].effect == EFFECT_PROTECT);
        PLAYER(SPECIES_WOBBUFFET);
```

File: test/battle/move_effect/coaching.c [hunk 3]
```diff
@@ -1,8 +1,112 @@
...
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_COACHING, playerLeft);
+        MESSAGE("Wynaut's Attack rose!");
+        MESSAGE("Wynaut's Defense rose!");
+    }
+}
+
+DOUBLE_BATTLE_TEST("Coaching bypasses Crafty Shield")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
```
Comment (hedara90) [inline test/battle/move_effect/coaching.c:39]:
```suggestion
    GIVEN {
        ASSUME(gMovesInfo[MOVE_CRAFTY_SHIELD].effect == EFFECT_PROTECT);
        PLAYER(SPECIES_WOBBUFFET);
```

File: test/battle/move_effect/coaching.c [hunk 4]
```diff
@@ -1,8 +1,112 @@
...
+        MESSAGE("Wynaut's Attack rose!");
+        MESSAGE("Wynaut's Defense rose!");
+    }
+}
+
+DOUBLE_BATTLE_TEST("Coaching fails if all allies are is semi-invulnerable")
+{
+    KNOWN_FAILING; // Coaching succeeds
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
```
Comment (hedara90) [inline test/battle/move_effect/coaching.c:56]:
```suggestion
    GIVEN {
        ASSUME(gMovesInfo[MOVE_FLY].effect == EFFECT_SEMI_INVULNERABLE);
        PLAYER(SPECIES_WOBBUFFET);
```

File: test/battle/move_effect/cosmic_power.c [hunk 5]
```diff
@@ -1,4 +1,16 @@
 #include "global.h"
 #include "test/battle.h"
```
Comment (hedara90) [inline test/battle/move_effect/cosmic_power.c:3]:
```suggestion

ASSUMPTIONS
{
    ASSUME(gMovesInfo[MOVE_COSMIC_POWER].effect == EFFECT_COSMIC_POWER);
}

```

**Review by hedara90 — COMMENTED** [review]
Missing some assumes from the tests.

**Comment by AsparagusEduardo** [comment]
Applied. Thank you for the fixes :)

**Review by hedara90 — APPROVED** [review]

**Pull request merged**